src/main/java/xbot/common/math/kinematics/DeadwheelKinematics.java

```diff
@@ -0,0 +1,146 @@
+package xbot.common.math.kinematics;
+
+import static edu.wpi.first.units.Units.Meters;
+
+import edu.wpi.first.math.MathSharedStore;
+import edu.wpi.first.math.MathUsageId;
+import edu.wpi.first.math.geometry.Twist2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.Kinematics;
+import edu.wpi.first.units.measure.Distance;
+
+/**
+ * Helper class that converts a deadwheel velocity (dx and dtheta components) to
+ * left and right wheel
+ * velocities for a differential drive.
+ *
+ * <p>
+ * Inverse kinematics converts a desired chassis speed into left and right
+ * velocity components
+ * whereas forward kinematics converts left and right component velocities into
+ * a linear and angular
+ * chassis speed.
+ */
+public class DeadwheelKinematics
+        implements Kinematics<DeadwheelWheelSpeeds, DeadwheelWheelPositions> {
+    /** Differential drive trackwidth. */
+    public final double robotWidthMeters;
+
+    /**
+     * Constructs a deadwheel kinematics object.
+     *
+     * @param robotWidthMeters The track width of the drivetrain. Theoretically,
+     *                         this is the distance
+     *                         between the left wheels and right wheels. However,
+     *                         the empirical value may be larger than
+     *                         the physical measured value due to scrubbing effects.
+     */
+    public DeadwheelKinematics(double robotWidthMeters) {
```

Comment:
nit: these argument names, like `robotWidthMeters`, seem like they should be `distanceBetweenDeadwheelsMeters`.